src/main/java/org/testng/Assert.java

```diff
@@ -1280,13 +1280,18 @@ public static void assertEqualsDeep(Set<?> actual, Set<?> expected, String messa
       }
     }
 
+    if(expected.size()>actual.size()){
```

Comment:
A couple of things on this 

1. Please format this code snippet.
2. This condition doesn't handle the scenario wherein `expected.size() < actual.size()`. So wouldn't we be better off by just checking `expected.size() != actual.size()` ?
3. Add a new test for the less than condition as well.

Reply:
#####Please find my response below @krmahadevan 
Let me know your thoughts.
1. Format completed
2. The original implementation of `assertEqualsDeep` is to compare a set of items between actual and expected, this is similar to stringContains for set, for more deatils refer 'https://github.com/cbeust/testng/issues/1917#issuecomment-435259490'
3. Added test for less than condition as well `arrayDeepInSetsAssertEquals`

Reply:
> The original implementation of assertEqualsDeep is to compare a set of items between actual and expected, this is similar to stringContains for set, if we are using expected.size() != actual.size() then that is similar to assertEquals where the actual should always be exactly to expected not more not less, for more details refer '#1917 (comment)'

I dont think that's the case. `assertEquals` merely relies on the `.equals()` method and is agnostic of the underlying type of the object. IMO size check has to be done (`expected.size() != actual.size()`) to ensure there's symmetry in the comparison. Two sets should fail deep equality if they dont match in size. Only if the size matches is when we should proceed further with checking the contents of the set.

Please see https://github.com/cbeust/testng/issues/1917#issuecomment-435562562

Also the tests dont check for the error message but merely check for the assertion failure (which is going to happen anyway). Can we please refactor both the tests (`expected > actual` and `expected < actual`) that specifically check for the error message as well ?

@juherr please let me know your thoughts

In a nutshell, I believe that deepEquals

Reply:
I agree. The 2 collections are not equals and one new test looks wrong.